### ISSUE TITLE
Can't add new images from smb share

### ISSUE THREAD

**Issue by seyfeb** [darktable-org/darktable#9740]
**Describe the bug/issue**

When trying to add images (either via `add to library..` or copy & import` from a network share no images or directory structure is shown. It is impossible to import any new images.

**To Reproduce**
1. Go to 'lighttable'
2. Click on 'add to library..'
3. Select 'add a custom place' to add network share
4. See no folders and images in the left and right panes

**Expected behavior**

Directory structure is visible and can be navigated. New images can be imported via the 'add to library' button.

**Screenshots**
![Screenshot_2021-08-05_22-32-15](https://user-images.githubusercontent.com/7623143/128417191-2b783f4e-787f-4c29-a749-eace86c26487.png)

**Platform**

* darktable version : 3.6.0
* OS                : , Linux version 5.11.0-25-generic 
* Linux - Distro    : Xubuntu 21.04
* Desktop           :  Xfce 

**Additional context**
I have an smb share in the network that I'm trying to access which worked before and can be accessed from the file browser without any problems.
When trying access from the darktable dialog the terminal logs 
```
(darktable:2402): GLib-GIO-CRITICAL **: 22:22:05.566: g_file_enumerator_next_file: assertion 'G_IS_FILE_ENUMERATOR (enumerator)' failed
```

**Comment by Mark-64** [comment]
Duplicate of #9407

**Comment by phweyland** [comment]
You should get some default list in "places"...
If you umount you share, do you get something ? (make sure you reset the "places").
How do you mount your share ?

**Comment by seyfeb** [comment]
When it wasn't working, I deleted the default places. I also tried resetting them, which brings `home` and `pictures` back. The share was auto-mounted to `/run/user/1000/gvfs/smb-share:server=freenas.local,share=sebastian/Bilder`. What works, is manually mounting the share to a more sane path (like `/media/pictures/`), so it seems to be related to the path, which may need to be properly escaped

**Comment by phweyland** [comment]
Good to know you have at least a work around.

Do you mean that your path is:
`/run/user/1000/gvfs/smb-share:server=freenas.local`
?
I don't know this syntax. There are at least two uncommon characters in it...
Could you point to me to some documentation to learn a bit more about that ?

**Comment by buckket** [comment]
Hey, same problem here when using GVfs. When mounting manually it works fine.

GVfs is pretty common for mounting Samba shares in GNOME. Here’s the documentation: https://wiki.gnome.org/Projects/gvfs/doc

> The package provides convenient FUSE mount for easy POSIX access to active GIO mounts. This is handled by the gvfsd-fuse process, daemon that is autostarted when gvfsd is spawned. The default mount point is **/run/user/\<UID\>/gvfs**, usually located on tmpfs, with a fallback to the old location ~/.gvfs when not available. On some systems, the user needs to be in the fuse group to be actually able to mount FUSE filesystems.

> The daemon creates flat structure of active GIO mounts at the topmost level, named by internal GMountSpec string representation **(in the form of protocol:arg1=val,arg2=val)**. This string is not guaranteed and may change in the future.

**Comment by phweyland** [comment]
Thanks for the links. 
A bit strange, gvfs seems to be integrated in glib, therefore should work transparently ...

**Comment by phweyland** [comment]
`(darktable:2402): GLib-GIO-CRITICAL **: 22:22:05.566: g_file_enumerator_next_file: assertion 'G_IS_FILE_ENUMERATOR (enumerator)' failed`

is the message above issued when the import dialog is opened or when the mount (as bilder shown above) is selected in "places" (assuming "bilder" is not already selected when one opens the import dialog) ?

I haven't gnome installed and I won't install it...
Would one of you accept to build a test branch, and report back the traces to help to solve the issue ?

**Comment by dada051** [comment]
3.8.0 same problem

**Comment by bcelasun** [comment]
I have exactly the same problem reported by seyfeb at the start of this thread.
Manjaro Linux, 64bit, rolling as of Jan 20th, 2022. Gnome version 41.2.
The problem is the same for version 3.6.1 and 3.8.0 of darktable. 
Everything works OK other the the buttons : "add to library" and "copy & import". Clicking on any of these stops darktable from responding and it had to be killed.
The error message is this:
(darktable:5664): GLib-GIO-CRITICAL **: 19:16:17.320: g_file_enumerator_next_file: assertion 'G_IS_FILE_ENUMERATOR (enumerator)' failed

I have no Samba share. There are mounted drives.
My system and all the other software runs smoothly with no problem otherwise. 

phweyland:
>is the message above issued when the import dialog is opened or when the mount (as bilder shown above) is selected in >"places" (assuming "bilder" is not already selected when one opens the import dialog) ?

When I click to "add to library" (or copy & import) while my drive storing the photos mounted, the dialog box opens but at that point no further operation can be made.
When I click to "add to library" (or copy & import) while all my attached drives are mounted, the dialog box does not appear and darktable freezes.

**Comment by phweyland** [comment]
I'm not an expert on linux mount, but it seems that is the point which makes difference.
@bcelasun could describe enough how your drive is mounted?, hoping you don't use gnome's mounting ...

**Comment by bcelasun** [comment]
> hoping you don't use gnome's mounting ...

Unfortunately, I do :( 
I just click on the drive and it gets mounted. 
Yet, it had always been like that before without problem...

**Comment by bcelasun** [comment]
Since others (also using Manjaro Linux) do not report a problem, I guess the problem may have something to do with my re-organizing (and/or renaming) the photo storage directories.

**Comment by phweyland** [comment]
I understand. darktable uses glib (gnome lib, isn't it ?) but for some reason gnome mounted drives are not seen. We still have to discover why.
Meanwhile, under linux there is a way to mount a samba drive, see [here](https://github.com/darktable-org/darktable/pull/9420#issuecomment-875963711).
And as far as I know there are other ways to mount drive than using gnome...
I don't use gnome, so I'm not able to reproduce these issues unfortunately.

**Comment by phweyland** [comment]
Anyone, using gnome, to make a try on gvfs mounted drive with #10937 ?

**Comment by buckket** [comment]
I just came around to test this on 3.8.1 which should have the fix mentioned here.

I can add location but it still shows as empty and logging the following error:
`g_file_enumerator_next_file: assertion 'G_IS_FILE_ENUMERATOR (enumerator)' failed`

Sometimes I even get a SIGSEGV when clicking on the folder.

**Comment by domi84** [comment]
Hello,
I just installed Ubuntu 20.04 LTS, the darktable version available is 3.8.1. When I tried to load the pics from my NAS (mounted easily from Gnome) I was getting the same error described here and darktable was crushing

```(darktable:2402): GLib-GIO-CRITICAL **: 22:22:05.566: g_file_enumerator_next_file: assertion 'G_IS_FILE_ENUMERATOR (enumerator)' failed```

I managed to solve it mounting via cifs https://wiki.ubuntu.com/MountWindowsSharesPermanently
Maybe it can helps others
